### ISSUE TITLE
Doxygen documentation

### ISSUE THREAD

**Issue by aydoganersoz** [aydoganersoz/tomat#7]

**Comment by ivajloip** [comment]
normally you would use `cargo doc` for your documentation and it has worked rather well for me. Do you really want `Doxygen` or you want some code documentation :slightly_smiling_face:

**Comment by aydoganersoz** [comment]
it was just to not forget about the documentation :slightly_smiling_face: thanks for your advice, I would always appreciate your further comments :slightly_smiling_face:

**Comment by aydoganersoz** [comment]
https://doc.rust-lang.org/stable/rust-by-example/testing/doc_testing.html

**Comment by ivajloip** [comment]
Yes, I have been using that in my docs - quite useful :slightly_smiling_face:
